build_tools/circle/build_doc.sh

```diff
@@ -51,5 +52,11 @@ source activate testenv
 # Build and install scikit-learn in dev mode
 python setup.py develop
 
+if [[ "$CIRCLE_BRANCH" =~ ^master$|^[0-9]+\.[0-9]+\.X$ ]]
```

Comment:
Indeed, from what I understand, when you do a PR, the build is done on the branch once. In this first build, we'll have CIRCLE_BRANCH = "name_of_the_branch_on_your_fork" and CI_PULL_REQUEST="something"
Then I think, when you merge it to master, this generates another build which will be build from merged master with CI_PULL_REQUEST = nothing. That's what I'm trying to check right now but I keep having time out errors and error from Travis for some reasons I don't get...